### ISSUE TITLE
Split examples into separate folders

### ISSUE THREAD

**Issue by bookercodes** [25th-floor/revalidation#22]
At the moment, `example` has a lot of "boilerplate" to accommodate multiple use-case demos, which can make the code harder than necessary to follow and understand.

Rather than having one `example` folder, let's create an `examples` folder with a sub-folder for each demo, e.g.:

```
+ examples
 | basic-form-validation
 | asynchronous-form-validation
 | password-form-validation
``` 
I propose we use [`react-scripts`](https://www.npmjs.com/package/react-scripts) to remove the need for `webpack.config.js` and other files not related to Revalidation. We should also use minimal styling in this folder to keep the examples focused on how Revalidation works 👍.

We could consider building a nice-looking demo for the website but keeping it separate from examples?

**Comment by busypeoples** [comment]
Yes, makes sense. We can add a styled example with Rebass or SC.

**Comment by bookercodes** [comment]
So, I am thinking these would be some good examples:


* Simple validation
* Asynchronous validation (with progress indicator)
* Password validation (comparing fields)
* Complex form (examples of `input[type="text"]` `select`, and other form elements)
* Disable submit button
* React Bootstrap example (good example of how easily you can add custom styles)
* Example with [validator.js](https://validatejs.org/#utilities-is-number)

What do you think? Also, what else can we add?

**Comment by busypeoples** [comment]
Sounds great. Maybe a rebass example? 
https://www.webpackbin.com/bins/-KnG58K1TEbUzSmRsmD5
